File: CS/dxSample_CustomSimpleButton/Custom Button/CustomButton.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Linq;
using DevExpress.LookAndFeel;
using DevExpress.Skins;
using DevExpress.XtraEditors;

namespace dxSample_CustomSimpleButton.Custom_Button {
    [ToolboxItem(true)]
    internal class CustomButton : SimpleButton {
        protected override DevExpress.XtraEditors.ViewInfo.BaseStyleControlViewInfo CreateViewInfo() {
            return new CustomSimpleButtonViewInfo(this);
        }
        private CustomStyle _CustomStyle = CustomStyle.Style1;
        public CustomButton() {
            AssignCustomImage();
        }
        public void AssignCustomImage() {
            var skin = GetCurrentSkin();
            var img = GetImage();
            if (img != null) {
                SetElementImage(skin, img);
            }
        }
        public CustomStyle CustomStyle {
            get {
                return _CustomStyle;
            }
            set {
                _CustomStyle = value;
            }
        }
        private Image GetImage() {
            return global::dxSample_CustomSimpleButton.Properties.Resources.NewButton;
        }
        public void SetElementImage(Skin currentSkin, Image image) {
            var element = currentSkin[CommonSkins.SkinButton];
            element.SetActualImage(image, false);
            element.Image.ImageCount = 18;
        }
        private Skin GetCurrentSkin() {
            return CommonSkins.GetSkin(UserLookAndFeel.Default.ActiveLookAndFeel);
        }
    }
}

File: VB/dxSample_CustomSimpleButton/Custom Button/CustomButton.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Linq
Imports DevExpress.LookAndFeel
Imports DevExpress.Skins
Imports DevExpress.XtraEditors

Namespace dxSample_CustomSimpleButton.Custom_Button
    <ToolboxItem(True)> _
    Friend Class CustomButton
        Inherits SimpleButton

        Protected Overrides Function CreateViewInfo() As DevExpress.XtraEditors.ViewInfo.BaseStyleControlViewInfo
            Return New CustomSimpleButtonViewInfo(Me)
        End Function
        Private _CustomStyle As CustomStyle = CustomStyle.Style1
        Public Sub New()
            AssignCustomImage()
        End Sub
        Public Sub AssignCustomImage()
            Dim skin = GetCurrentSkin()
            Dim img = GetImage()
            If img IsNot Nothing Then
                SetElementImage(skin, img)
            End If
        End Sub
        Public Property CustomStyle() As CustomStyle
            Get
                Return _CustomStyle
            End Get
            Set(ByVal value As CustomStyle)
                _CustomStyle = value
            End Set
        End Property
        Private Function GetImage() As Image
            Return My.Resources.NewButton
        End Function
        Public Sub SetElementImage(ByVal currentSkin As Skin, ByVal image As Image)
            Dim element = currentSkin(CommonSkins.SkinButton)
            element.SetActualImage(image, False)
            element.Image.ImageCount = 18
        End Sub
        Private Function GetCurrentSkin() As Skin
            Return CommonSkins.GetSkin(UserLookAndFeel.Default.ActiveLookAndFeel)
        End Function
    End Class
End Namespace
